### ISSUE TITLE
feature request: backup/load native Mixxx library

### ISSUE THREAD

**Issue by deejay-girl** [mixxxdj/mixxx#12120]
### Feature Description

When reading this request https://github.com/mixxxdj/mixxx/issues/12117 I noticed there is no way to backup a Mixxx library in case you want to transfer it to another computer. At least not in a user friendly way by manually copying files.
So there must be a user friendly way to do this, from within Mixxx itself.

I would like to request a feature in the "library" menu:

(1) backup Mixxx library: this asks the user where to save the Mixxx library to 1 file which has everything inside
(2) load Mixxx library: this asks the user to load a previously backup of a Mixxx library (as in (1))

Of course the backup must work across platforms.
I'm not a developer so I don't know what format the file should have.

**Comment by deejay-girl** [comment]
Proof people need this: https://mixxx.discourse.group/t/synchronizing-library-across-multiple-computers/27937

**Comment by fwcd** [comment]
Yeah, this is an area that definitely could see some improvement. I've tried implementing such a backup workflow as an [external program](https://github.com/fwcd/mixync) in the past, but having a native solution would of course be preferable. IIRC, @xeruf built something similar a while ago: https://github.com/xeruf/mixxxdb-kt

I would actually go a step further and propose that the Mixxx library itself should be portable (i.e. split out the music library part, i.e. crates, playlists, tracks, ...) from the settings that are currently also stored in the Mixxx database, so people can just keep it alongside their music, including on external drives etc. There are a lot of design questions around that though. If you're curious, have a look at:

- https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Portable.20Library.20Folders/near/388723120
- https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Portable.20library
